### PR TITLE
Thchang/file browser progress bar

### LINKED ISSUE #1303: support progress bar and cancellation in the file browser
to support https://github.com/CARTAvis/carta-backend/issues/431 the frontend needs a progress bar and cancellation button while the backend is parsing image types from a folder with lots files.

### PR DESCRIPTION
**Pull request by TienHao**
closes #1303 to support backend issue(CARTAvis/carta-backend#431). 

Add progress bar with cancel button in the file browser. The file browser will show the progress bar after receiving a progress message from the backend. The cancel button will cancel the file list loading process and return to the previous directory.

companion backend branch: [mark/431_file_list_interruption](https://github.com/CARTAvis/carta-backend/tree/mark/431_file_list_interruption) (CARTAvis/carta-backend#777)
companion protobuf branch: [mark/stop_getting_file_list](https://github.com/CARTAvis/carta-protobuf/tree/mark/stop_getting_file_list)

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
Feature-wise this looks good. 

One issue I spotted (not due to this PR) is that when we load _another_ image from a folder with lots of files, initially we will see the file list saying "Empty folder" before the long file list arrives. Will file this as a new issue. 


https://user-images.githubusercontent.com/20819712/116189706-ef162300-a75b-11eb-8e4c-77857c2c6c28.mp4

File: src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx [hunk 1]
```diff
@@ -152,6 +152,16 @@ export class FileBrowserDialogComponent extends React.Component {
         fileBrowserStore.setExportFilename(ev.target.value);
     };
 
+    private handleFileBrowserRequestingCancelled = () => {
```
Comment (veggiesaurus) [inline src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx]: rename to `handleFileBrowserRequestCancelled`

File: src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx [hunk 2]
```diff
@@ -546,6 +556,15 @@ export class FileBrowserDialogComponent extends React.Component {
...
                     This file exists. Are you sure to overwrite it?
                 </Alert>
+                <TaskProgressDialogComponent
+                    isOpen={fileBrowserStore.isLoadingDialogOpen && fileBrowserStore.loadingProgress < 1}
+                    progress={fileBrowserStore.loadingProgress}
+                    timeRemaining={appStore.estimatedTaskRemainingTime}
+                    cancellable={true}
+                    onCancel={this.handleFileBrowserRequestingCancelled}
+                    text={"Loading files"}
+                    contentText={`loading files ${fileBrowserStore.loadingCheckedCount} / ${fileBrowserStore.loadingTotalCount}`}
```
Comment (veggiesaurus) [inline src/components/Dialogs/FileBrowser/FileBrowserDialogComponent.tsx]: Should this be `Loading`?

**Review by veggiesaurus — CHANGES_REQUESTED** [review]
I've requested changes on the protobuf repo itself, which will affect both frontend and backend PRs

**Comment by veggiesaurus** [comment]
@TienHao will you update this branch to make use of the new protobuf approach @markccchiang has implemented?

**Comment by TienHao** [comment]
merged with dev and updated with new protobuf

File: src/stores/FileBrowserStore.ts [hunk 3]
```diff
@@ -369,6 +375,36 @@ export class FileBrowserStore {
...
+        this.loadingTotalCount = totalCount;
+    };
+
+    @action resetLoadingStates = () => {
+        this.loadingList = false;
+        this.isLoadingDialogOpen = false;
+        this.updateLoadingState(0, 0, 0);
+    };
+
+    @action cancelRequestingFile = () => {
```
Comment (veggiesaurus) [inline src/stores/FileBrowserStore.ts]: can't we just combine this into one function? `this.browserMode` can be used to determine the `FileListType`

**Review by veggiesaurus — CHANGES_REQUESTED** [review]

**Review by veggiesaurus — APPROVED** [review]

**Pull request merged**